### DISCUSSION TITLE
A few questions about images

### DISCUSSION THREAD

**Discussion by upcomingtradera** [razonyang/hugo-theme-bootstrap#945] [Q&A]
in this [image section](https://hbs.razonyang.com/v1/en/docs/content/#thumbnail-selection-order) about thumbnails, feature images, etc...

You want the images in the front matter. I can add 1 image and it will be the main image for the blog post if I add

images = [/images/image.jpg']

Let's say that I have my posts in content/posts/post-title.md
i place all my images in static/images/post-title.jpg

then I add this to the front matter of my toml file
images = ['/images/post-title.jpg']

if I add these two image files to my static/image folder
post-title-feature.jpg
post-title-thumbnail.jpg

then i add both to my post images section
images = ['/images/post-title-feature.jpg', '/images/post-title-thumbnail.jpg']

only one show up as both the featured image and the thumbnail.

can you explain the feature, thumbnail and cover image and how do I properly setup and use them?

I have all my posts as .md files in content/posts directory and didn't create subfolder for each post. Do I need to create a sub folder for each post to have these image types work correctly?

**Comment by upcomingtradera** [comment]
I was editing and realized I need to use the new content bundle, okay so I sorted that but there's featured image which is above the fold when you open the blog post. A cover image which is the smaller image on the blog roll page.

Where does the thumbnail image come show up?

**Comment by upcomingtradera** [comment]
what is the order if i have:
feature.jpg
cover.jpg
thumbnail.jpg

can i have all three and they all show up? It seems like if feature is there, nothing else shows up. If no feature then cover shows up. I haven't seen thumbnail show up.

Can you provide some clarification on those?

**Comment by razonyang** [comment]
The featured image is designed to show only one image.

1) If the `images` is present on the front matter, then the _first_ image will be used as the featured image.
2) If the `images` is not set, then the theme will look for the _page image resources_ which filename match the patterns: `*feature*`. Btw, the _page image resources_ are images that inside the page. Take page `content/foo/index.md` as an example, `content/foo/featured.png` will be picked as the feature image.

The theme will try to resize the page image resources (not `/static` images) to multiple sizes, you don't have to take care of the thumbnail size.

According to the example you provided, you're misunderstanding the page resources, the images should be placed in the page folder. The structure like this:

```
$ tree content/foo
├── featured.png
├── index.md
```

**Reply by upcomingtradera** [reply]
it seemed like the theme wasn't properly picking up the localresource images and the links would break.

I made a few changes with here: https://github.com/upcomingtradera/hugo-theme-bootstrap/commit/ac9a1cbcbe93d6c0c958af6c7fc7dc9d5351cb1a

i put featured_image and thumbnail_image as their own array in the front matter
there's also featured_image_alt_text and thumbnail_alt_text which allows you to specify a different alt text or it defaults to the title of the post.

It also allows you to have a simpler way to have a different thumbnail vs featured image. Although the main issue was that even with the resource bundle format, the site as it is right now will not link the images properly. If i remember correctly you needed to set permalink and not relpermalink.

**Reply by razonyang** [reply]
> it seemed like the theme wasn't properly picking up the localresource images and the links would break.

The theme follows the same rule of some Hugo built-in template, such as `open-graph` and `twitter-card`.

**Comment by upcomingtradera** [comment]
there are some issues with the way you handle image resources.

**Reply by razonyang** [reply]
Could you please describe in details? It would be great if you can provide examples for reproducing.

**Comment by upcomingtradera** [comment]
I think the way you handle images is missing a few features. Remember that images can be bundled resources or other formats such as
page resources
global resources
page bundle
static resources

check to make sure that your processing handles all of those cases.
